fdbcli/fdbcli.actor.cpp

```diff
@@ -2296,59 +2297,59 @@ ACTOR Future<bool> exclude( Database db, std::vector<StringRef> tokens, Referenc
 			warn.cancel();
 
 		state std::set<NetworkAddress> notExcludedServers =
```

Comment:
@xumengpanda We could change this to `std::set<AddressExclusion>` as you suggested, but we'd need to be careful to populate it with the matching exclusions and not new exclusions constructed from worker ip addresses. Which do you prefer?

Reply:
I'm a bit lost at `not new exclusions constructed from worker ip addresses. `.  Could you please give an example?

Reply:
Currently checkForExcludingServers constructs its result from addresses of workers under `\xff/serverList/`. Since addresses of workers always have a port, they'll never compare equal to a whole-machine exclusion (and consequently, `exclude no_wait` would always think a whole-machine exclusion is complete) . If we returned the matching exclusions instead of the matching workers from checkForExcludingServers then we could use notExcludedServers the way it's currently in use.

Reply:
It seems like the source of the mismatch is the `NetworkAddress`s persisted in private key space for the storage servers could have their TLS flag set, while `fdbcli` explicitly ask the operator NOT to append `:tls` for the excluding address and thus the `NetworkAddress` parsed out would never have the TLS flag set.

@vishesh Do you still remember why we decided not to support TLS here?

Reply:
After a brief look I think `AddressExclustion` never had to worry about TLS. So I think Andrew's suggestion is the best solution.

Reply:
I feel that exclusion is based on ip:port or ip. Flags are not expected to be passed around in the exclude logic. 
It seems that the problem with excluding an IP exists for either `NetworkAddress ` or `AddressExclusion`. 

I don't have a strong opinion on this.